core/src/main/kotlin/xtdb/api/MetricsConfigFactory.kt

```diff
@@ -0,0 +1,10 @@
+package xtdb.api
+
+import io.micrometer.core.instrument.MeterRegistry
+
+interface MetricsConfigFactory {
```

Comment:
naming: `MetricsRegistryFactory`? `ConfigFactory` implies we're returning an object that'll help build the config, which I don't think is quite right?